lalrpop/src/grammar/parse_tree.rs

```diff
@@ -149,10 +149,12 @@ pub struct NonterminalData {
     pub alternatives: Vec<Alternative>
 }
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Annotation {
     pub id_span: Span,
     pub id: InternedString,
+    // TODO nixpulvis: This is not exactly the correct type.
+    pub parameters: Vec<(InternedString, InternedString)>,
```

Comment:
I'm all for a grammar for annotations, I'm not exactly sure what you mean by parsing the Debug output but it sounds hacky.